pkgs/development/python-modules/berkeleydb/default.nix

```diff
@@ -19,6 +20,8 @@ buildPythonPackage rec {
 
   build-system = [ setuptools ];
 
+  doCheck = !stdenv.isDarwin;
```

Comment:
Please note what has failed on darwin (and possible reason, if you have)

Reply:
It looks like all the tests fail quite comprehensively on Hydra. Are we sure this package actually works okay on Darwin?

Reply:
Updated with the error 👍

```
python3.12-berkeleydb> Traceback (most recent call last):
python3.12-berkeleydb>   File "/nix/store/9pj4rzx5pbynkkxq1srzwjhywmcfxws3-python3-3.12.5/lib/python3.12/unittest/case.py", line 58, in testPartExecutor
python3.12-berkeleydb>     yield
python3.12-berkeleydb>   File "/nix/store/9pj4rzx5pbynkkxq1srzwjhywmcfxws3-python3-3.12.5/lib/python3.12/unittest/case.py", line 630, in run
python3.12-berkeleydb>     self._callSetUp()
python3.12-berkeleydb>   File "/nix/store/9pj4rzx5pbynkkxq1srzwjhywmcfxws3-python3-3.12.5/lib/python3.12/unittest/case.py", line 586, in _callSetUp
python3.12-berkeleydb>     self.setUp()
python3.12-berkeleydb>   File "/private/tmp/nix-build-python3.12-berkeleydb-18.1.10.drv-1/berkeleydb-18.1.10/tests/test_weakref.py", line 71, in setUp
python3.12-berkeleydb>     self.dbenv.open(self.homeDir, db.DB_CREATE | db.DB_INIT_MPOOL |
python3.12-berkeleydb> berkeleydb.db.DBRunRecoveryError: (-30973, 'BDB0087 DB_RUNRECOVERY: Fatal error, run database recovery -- BDB1546 unable to join the environment')
```

The error looks like an issue with how the test environment is configured on macOS for `berkeleydb`. I'm not sure if the `berkeleydb` library works on macOS as I don't have any files to test with through Gramps, however at the very least it imports successfully:

```
$ gramps --version
Databases:
-------------------------
 bsddb     :
     version     : 18.1.10
     db version  : 5.3.28
     location    : /nix/store/rjshbrg57hhyrc30xjl71l7xv2papcnj-python3.12-berkeleydb-18.1.10/lib/python3.12/site-packages/berkeleydb/__init__.py
```

The relevant source code is here:

https://github.com/gramps-project/gramps/blob/e84b26dddf4c6da66ae1b9b5271eacbc4971e500/gramps/grampsapp.py#L455-L469